Furball.Vixie/Graphics/Backends/Veldrid/LineRendererVeldrid.cs

```diff
@@ -0,0 +1,223 @@
+using System;
+using System.Numerics;
+using System.Reflection.Metadata;
+using System.Runtime.InteropServices;
+using System.Text;
+using Furball.Vixie.Graphics.Backends.Veldrid.Abstractions;
+using Furball.Vixie.Graphics.Renderers;
+using Veldrid;
+using Veldrid.SPIRV;
+
+namespace Furball.Vixie.Graphics.Backends.Veldrid {
+    public class LineRendererVeldrid : ILineRenderer {
+        private readonly VeldridBackend _backend;
+        
+        private Pipeline _pipeline;
+        
+        [StructLayout(LayoutKind.Sequential)]
+        private struct InstanceData {
+            public Vector2 InstancePosition;
+            public Vector2 InstanceSize;
+            public Color   InstanceColor;
+            public float   InstanceRotation;
+        }
+        
+        
+        [StructLayout(LayoutKind.Sequential)]
+        private struct Vertex {
+            public Vector2 VertexPosition;
+        }
+
+        private readonly DeviceBuffer _projectionBuffer;
+        private readonly ResourceSet  _projectionBufferResourceSet;
+        
+        private readonly DeviceBuffer _instanceVertexBuffer;
+        private readonly DeviceBuffer _vertexBuffer;//TODO: maybe this can be static?
```

Comment:
why would you want it static

Reply:
because it never changes

Reply:
readonly does that already, if multiple line renderers exist itll just fuck stuff up

Reply:
> readonly does that already, if multiple line renderers exist itll just fuck stuff up

No as in, the buffer is never changed from its initial value, only the instance vertex buffer changes, we could use the same vertex buffer for multiple line renderers, and this applies to quad renderers too

Reply:
Itd save a small amount of VRAM

Reply:
id still rather have this be readonly and have that be it, that miniscule vram waste aint the end of the world, even on mobile

Reply:
i mean i guess